Lib/test/test_weakref.py

```diff
@@ -101,6 +101,11 @@ def collect():
             t.join()
 
 
+skip_if_tsan_and_gil_disabled = unittest.skipIf(
+    support.check_sanitizer(thread=True) and support.Py_GIL_DISABLED,
+    "Test is prohibitively slow with TSAN enabled and the GIL disabled")
```

Comment:
Is it possible to just reduce the iteration count in this case instead? Ideally we shouldn't skip threaded stress tests on free-threading builds :-)

Reply:
We have to reduce the iteration count by a couple orders of magnitude, but that should work.